Change `AssetHierarchy` validation: Assume parent ID linked assets are valid (#1283)
cognitedata/cognite-sdk-python: CHANGELOG.md

- `Fixed` for any bug fixes.
- `Security` in case of vulnerabilities.

## [6.8.1] - 2023-07-08
### Changed

> - The `AssetHierarchy` class would consider assets linking its parent only by ID as orphans, contradicting the
>   docstring stating "All assets linking a parent by ID are assumed valid". This is no longer the case.

Author: ```suggestion
- The `AssetHierarchy` class would consider assets linking their parent only by ID as orphans, contradicting the
  docstring stating "All assets linking a parent by ID are assumed valid". This is now true (they are no longer
  considered orphans).
```
